store.go

```diff
@@ -94,37 +101,33 @@ func NewStore(dbFile string, options ...StoreOption) (Store, error) {
 	return st, nil
 }
 
-func (s *store) JSONCollection(name string) Collection {
+func (s *store) Collection(collectionType CollectionType, name string) Collection {
 	c, ok := s.collections[name]
 	if !ok {
-		c = &collection{
-			name:           name,
-			db:             s.db,
-			refMake:        defaultReferenceCreator,
-			valueCollector: JSONPathValueCollector,
+		var vCollector valueCollector
+		switch collectionType {
+		case JSONCollection:
+			vCollector = JSONPathValueCollector
+		case JSONLDCollection:
+			vCollector = JSONLDValueCollector
+		default:
+			panic("unknown collection type")
 		}
-		s.collections[name] = c
-	}
-
-	return c
-}
-
-func (s *store) JSONLDCollection(name string) Collection {
-	c, ok := s.collections[name]
-	if !ok {
 		c = &collection{
 			name:           name,
+			collectionType: collectionType,
 			db:             s.db,
-			refMake:        defaultReferenceCreator,
 			documentLoader: s.documentLoader,
-			valueCollector: JSONLDValueCollector,
+			refMake:        defaultReferenceCreator,
+			valueCollector: vCollector,
 		}
 		s.collections[name] = c
+	} else if c.collectionType != collectionType {
+		panic("collection already exists with different type")
```

Comment:
I hate libraries that panic

Reply:
It would be a programming error, can't happen runtime unless you make really really stupid code. Therefor adding an error check creates code that will be never run.